<img src=".\images\Logo.png" width=150 align="left" /> <img src=".\images\Logo2.jpg" width=450 align="right" />


# <center><font color= #1e8449 > <b>CUxx_xxxxx</font></center>

<font color='steelblue'><b>Citizenlab Data Science Methodology > II - Data Processing Domain </font>
***
> # <font color='steelblue'> <b>05.- Data Collection</font>

<font color='steelblue'>Data Collection is the process to obtain and generate (if required) necessary data to model the problem.</font>

### 02. Crear csv con datos de infraestructuras

* Consolidar los datos en formato shp en data.frames con coordenadas
* Unir datos de hospitales de cu_25
* Total 12 ficheros de entrada y uno de salida







<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Settings" data-toc-modified-id="Settings-1"><font color="green">Settings</font></a></span></li><li><span><a href="#Data-Load" data-toc-modified-id="Data-Load-2"><font color="green">Data Load</font></a></span></li><li><span><a href="#ETL-Processes" data-toc-modified-id="ETL-Processes-3">ETL Processes</a></span><ul class="toc-item"><li><span><a href="#Import-data-from:-CSV,-Excel,-Tab,-JSON,-SQL,-and-Parquet-files" data-toc-modified-id="Import-data-from:-CSV,-Excel,-Tab,-JSON,-SQL,-and-Parquet-files-3.1">Import data from: CSV, Excel, Tab, JSON, SQL, and Parquet files</a></span></li></ul></li><li><span><a href="#Synthetic-Data-Generation" data-toc-modified-id="Synthetic-Data-Generation-4">Synthetic Data Generation</a></span></li><li><span><a href="#Fake-Data-Generation" data-toc-modified-id="Fake-Data-Generation-5">Fake Data Generation</a></span></li><li><span><a href="#Open-Data" data-toc-modified-id="Open-Data-6">Open Data</a></span></li><li><span><a href="#Data-Save" data-toc-modified-id="Data-Save-7"><font color="green">Data Save</font></a></span></li><li><span><a href="#Main-Conclusions" data-toc-modified-id="Main-Conclusions-8"><font color=" #2874a6 ">Main Conclusions</font></a></span></li><li><span><a href="#Main-Actions" data-toc-modified-id="Main-Actions-9"><font color=" #2874a6 ">Main Actions</font></a></span><ul class="toc-item"><li><ul class="toc-item"><li><span><a href="#Acciones-done" data-toc-modified-id="Acciones-done-9.0.1"><font color=" #2874a6 ">Acciones done</font></a></span></li><li><span><a href="#Acctions-to-perform" data-toc-modified-id="Acctions-to-perform-9.0.2"><font color=" #2874a6 ">Acctions to perform</font></a></span></li></ul></li></ul></li></ul></div>

## <font color='green'>Settings</font>

### Packages to use

- {sf} para datos georeferenciados
- {dplyr} para exploración y manipulación de datos
- {readr} para leer y escribir archivos csv


In [1]:
library(sf)
library(dplyr)
library(readr)


Linking to GEOS 3.10.2, GDAL 3.4.2, PROJ 8.2.1; sf_use_s2() is TRUE


Attaching package: ‘dplyr’


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union




### Paths

In [2]:
iPath <- "Data/Input/"
oPath <- "Data/Output/"

## <font color='green'>Data Load</font>

If there are more than one input file, make as many sections as files to import.

Instrucciones
- Los ficheros de entrada del proceso están siempre en Data/Input/.   
- Si hay más de un fichero de entrada, se crean tantos objetos iFile_xx y file_data_xx como ficheros de entrada (xx número correlativo con dos dígitos, rellenar con ceros a la izquierda)

### 1. Intercambiadores

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 


Data load using the {tcltk} package. Ucomment the line if not using this option

In [ ]:
# file_data_01 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>

NOTA: se omiten las comprobaciones de existencia de archivo para mejor compresión del código

In [6]:
iFile_01 <- "Transportes y Comunicaciones_ Intercambiadores/intercambiadores.shp"
file_data_01 <- paste0(iPath, iFile_01)


#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [8]:
data_01 <- st_read(file_data_01,
                            options = "ENCODING=latin1", 
                            quiet = FALSE)


options:        ENCODING=latin1 
Reading layer `intercambiadores' from data source 
  `/Users/emilio.lcano/academico/gh_repos/__transferencia/citizenlab/CitizenLab-Research-and-Development/casos_urjc/notebooks/II_data_processing/18_infraestructuras/Data/Input/Transportes y Comunicaciones_ Intercambiadores/intercambiadores.shp' 
  using driver `ESRI Shapefile'
Simple feature collection with 24 features and 8 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 434246 ymin: 4466946 xmax: 451488 ymax: 4482485
Projected CRS: ETRS89 / UTM zone 30N


Estructura de  los datos:

In [9]:
glimpse(data_01)

Rows: 24
Columns: 9
$ CMUN      <chr> "079", "079", "079", "079", "079", "079", "079", "079", "079…
$ ETIQUETA  <chr> "Grandes Intercambiadores Plaza Elíptica", "Grandes Intercam…
$ LOCALIDAD <chr> "Madrid", "Madrid", "Madrid", "Madrid", "Madrid", "Madrid", …
$ DIRECCION <chr> "Plaza Elíptica s/n", "Calle Princesa, 89", "Estación Prínci…
$ CODIGO2   <int> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 1…
$ UTM_X     <dbl> 439178, 438974, 438957, 441586, 442605, 442151, 441253, 4424…
$ UTM_Y     <dbl> 4470780, 4476259, 4474712, 4479841, 4476589, 4480368, 447757…
$ COD_SUCA  <chr> "07902184", "07904939", "07905571", "07905377", "07900312", …
$ geometry  <POINT [m]> POINT (439178.2 4470780), POINT (438974 4476259), POIN…


Muestra de datos:

In [15]:
data_01 |> slice_head(n = 5)

CMUN ETIQUETA                                    LOCALIDAD
1 079  Grandes Intercambiadores Plaza Elíptica     Madrid   
2 079  Grandes Intercambiadores Moncloa            Madrid   
3 079  Grandes Intercambiadores Príncipe Pío       Madrid   
4 079  Grandes Intercambiadores Plaza de Castilla  Madrid   
5 079  Grandes Intercambiadores Avenida de América Madrid   
  DIRECCION                                CODIGO2 UTM_X  UTM_Y   COD_SUCA
1 Plaza Elíptica s/n                       1       439178 4470780 07902184
2 Calle Princesa, 89                       2       438974 4476259 07904939
3 Estación Príncipe Pío                    3       438957 4474712 07905571
4 Calle San Aquilino y Avenida de Asturias 4       441586 4479841 07905377
5 Avenida de América 9 A                   5       442605 4476589 07900312
  geometry                
1 POINT (439178.2 4470780)
2 POINT (438974 4476259)  
3 POINT (438957 4474712)  
4 POINT (441586 4479841)  
5 POINT (442605 4476589)

### 2. Metro

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 


Data load using the {tcltk} package. Ucomment the line if not using this option

In [17]:
# file_data_02 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>

NOTA: se omiten las comprobaciones de existencia de archivo para mejor compresión del código

In [20]:
iFile_02 <- "Transportes y comunicaciones_ Metro (bocas)/"
file_data_02 <- paste0(iPath, iFile_02)


#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [21]:
data_02 <- st_read(file_data_02,
                            options = "ENCODING=latin1", 
                            quiet = FALSE)


options:        ENCODING=latin1 
Reading layer `bocas' from data source 
  `/Users/emilio.lcano/academico/gh_repos/__transferencia/citizenlab/CitizenLab-Research-and-Development/casos_urjc/notebooks/II_data_processing/18_infraestructuras/Data/Input/Transportes y comunicaciones_ Metro (bocas)' 
  using driver `ESRI Shapefile'
Simple feature collection with 771 features and 11 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 423308.6 ymin: 4459431 xmax: 461969.4 ymax: 4490120
Projected CRS: ETRS89 / UTM zone 30N


Estructura de  los datos:

In [22]:
glimpse(data_02)

Rows: 771
Columns: 12
$ ESTACION   <chr> "Plaza de Castilla", "Plaza de Castilla", "Plaza de Castill…
$ LINEAS     <chr> "1, 10, 9", "1, 10, 9", "1, 10, 9", "1", "1", "1", "1", "1"…
$ SALIDA     <chr> "Castellana", "Plaza de Castilla", "Pº Castellana, impares"…
$ DIRECCION  <chr> "Frente depósito C. Isabel II", "Paseo de la Castellana, bu…
$ MUNICIPIO  <chr> "Madrid", "Madrid", "Madrid", "Madrid", "Madrid", "Madrid",…
$ COD_MUNI   <chr> "079", "079", "079", "079", "079", "079", "079", "079", "07…
$ HORARIO    <chr> "Abierto de 6:00 a 1:30, a excepción de los accesos con hor…
$ MECANIZADO <chr> "Mecanizado: 6:00 a 13:50, S,D, Festivo y Verano de 2/6 a 2…
$ UTM_X      <dbl> 441600, 441574, 441545, 441012, 441161, 440713, 440830, 440…
$ UTM_Y      <dbl> 4479652, 4479802, 4479841, 4479502, 4479577, 4479032, 44791…
$ CODIGO2    <int> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, …
$ geometry   <POINT [m]> POINT (441599.6 4479652), POINT (441573.6 4479802), P…


Muestra de datos:

In [23]:
data_02 |> slice_head(n = 5)

ESTACION          LINEAS   SALIDA                
1 Plaza de Castilla 1, 10, 9 Castellana            
2 Plaza de Castilla 1, 10, 9 Plaza de Castilla     
3 Plaza de Castilla 1, 10, 9 Pº Castellana, impares
4 Valdeacederas     1        Capitán Blanco Argibay
5 Valdeacederas     1        Aníbal                
  DIRECCION                               MUNICIPIO COD_MUNI
1 Frente depósito C. Isabel II            Madrid    079     
2 Paseo de la Castellana, bulevar central Madrid    079     
3 Paseo Castellana, 189 (junto a torres)  Madrid    079     
4 Calle Bravo Murillo, 324                Madrid    079     
5 Calle Bravo Murillo, 350                Madrid    079     
  HORARIO                                                                
1 Abierto de 6:00 a 1:30, a excepción de los accesos con horario especial
2 Abierto de 6:00 a 1:30, a excepción de los accesos con horario especial
3 Abierto de 6:00 a 1:30, a excepción de los accesos con horario especial
4 Abierto de 6:00 a 1:30, a excepción de los accesos con horario especial
5 Abierto de 6:00 a 1:30, a excepción de los accesos con horario especial
  MECANIZADO                                                    UTM_X  UTM_Y  
1 Mecanizado: 6:00 a 13:50, S,D, Festivo y Verano de 2/6 a 29/9 441600 4479652
2 NA                                                            441574 4479802
3 NA                                                            441545 4479841
4 NA                                                            441012 4479502
5 Mecanizado permanente                                         441161 4479577
  CODIGO2 geometry                
1 1       POINT (441599.6 4479652)
2 2       POINT (441573.6 4479802)
3 3       POINT (441544.6 4479841)
4 4       POINT (441011.6 4479502)
5 5       POINT (441160.6 4479577)

### 3. Aeropuertos

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 


Data load using the {tcltk} package. Ucomment the line if not using this option

In [ ]:
# file_data_03 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>

NOTA: se omiten las comprobaciones de existencia de archivo para mejor compresión del código

In [24]:
iFile_03 <- "Transportes y comunicaciones_ Aeropuertos/"
file_data_03 <- paste0(iPath, iFile_03)


#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [25]:
data_03 <- st_read(file_data_03,
                            options = "ENCODING=latin1", 
                            quiet = FALSE)


options:        ENCODING=latin1 
Reading layer `aeropuertos' from data source 
  `/Users/emilio.lcano/academico/gh_repos/__transferencia/citizenlab/CitizenLab-Research-and-Development/casos_urjc/notebooks/II_data_processing/18_infraestructuras/Data/Input/Transportes y comunicaciones_ Aeropuertos' 
  using driver `ESRI Shapefile'
Simple feature collection with 3 features and 7 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 434226.4 ymin: 4469153 xmax: 462119.5 ymax: 4480754
Projected CRS: ETRS89 / UTM zone 30N


Estructura de  los datos:

In [26]:
glimpse(data_03)

Rows: 3
Columns: 8
$ CMUN      <chr> "079", "065", "148"
$ BUSCA     <chr> "Aeropuerto Adolfo Suárez-Madrid Barajas", "Aeropuerto Madri…
$ DIRECCIÓN <chr> "Avda. Hispanidad, s/n", "Ctra. Barrio de la Fortuna, s/n", …
$ MUNICIPIO <chr> "MADRID", "MADRID", "TORREJÓN DE ARDOZ"
$ CODIGO2   <int> 1, 2, 3
$ ETIQUETA  <chr> "Adolfo Suárez-Madrid Barajas", "Madrid-Cuatro Vientos", "Ma…
$ INFOR     <chr> "Madrid-Barajas<br>Avda. Hispanidad, s/n", "Madrid-Cuatro Vi…
$ geometry  <POINT [m]> POINT (451504.3 4478985), POINT (434226.4 4469153), POINT (4…


Muestra de datos:

In [27]:
data_03 |> slice_head(n = 5)

CMUN BUSCA                                   DIRECCIÓN                      
1 079  Aeropuerto Adolfo Suárez-Madrid Barajas Avda. Hispanidad, s/n          
2 065  Aeropuerto Madrid-Cuatro Vientos        Ctra. Barrio de la Fortuna, s/n
3 148  Aeropuerto Madrid-Torrejón de Ardoz     Ctra. de la Base, s/n          
  MUNICIPIO         CODIGO2 ETIQUETA                    
1 MADRID            1       Adolfo Suárez-Madrid Barajas
2 MADRID            2       Madrid-Cuatro Vientos       
3 TORREJÓN DE ARDOZ 3       Madrid-Torrejón             
  INFOR                                                   
1 Madrid-Barajas<br>Avda. Hispanidad, s/n                 
2 Madrid-Cuatro Vientos<br>Ctra. Barrio de la Fortuna, s/n
3 Madrid-Torrejón<br>Ctra. de la Base, s/n                
  geometry                
1 POINT (451504.3 4478985)
2 POINT (434226.4 4469153)
3 POINT (462119.5 4480754)

### 4. Estaciones de Cercanías

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 


Data load using the {tcltk} package. Ucomment the line if not using this option

In [ ]:
# file_data_04 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>

NOTA: se omiten las comprobaciones de existencia de archivo para mejor compresión del código

In [28]:
iFile_04 <- "Transportes y comunicaciones_ Cercanías (estaciones)/"
file_data_04 <- paste0(iPath, iFile_04)


#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [96]:
data_04 <- st_read(file_data_04,
                            options = "ENCODING=latin1", 
                            quiet = FALSE)


options:        ENCODING=latin1 
Reading layer `estcerca' from data source 
  `/Users/emilio.lcano/academico/gh_repos/__transferencia/citizenlab/CitizenLab-Research-and-Development/casos_urjc/notebooks/II_data_processing/18_infraestructuras/Data/Input/Transportes y Comunicaciones_ Cercanías (estaciones)' 
  using driver `ESRI Shapefile'
Simple feature collection with 93 features and 8 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 900524.4 ymin: 4445522 xmax: 982989.2 ymax: 4531246
Projected CRS: ETRS89 / UTM zone 29N


Estructura de  los datos:

In [97]:
glimpse(data_04)

Rows: 93
Columns: 9
$ LINEA     <chr> "C-2, C-7, C-8", "C-2, C-7, C-8", "C-2, C-7, C-8", "C-2, C-7…
$ MUNICIPIO <chr> "Alcalá de Henares", "Torrejón de Ardoz", "Coslada", "Coslad…
$ CMUN      <chr> "005", "148", "049", "049", "079", "079", "079", "079", "079…
$ CONEXION  <chr> "Aparcamiento", "Aparcamiento", "Aparcamiento", "Aparcamient…
$ ETIQUETA  <chr> "Alcalá de Henares", "Torrejón de Ardoz", "San Fernando de H…
$ UTM_X     <int> 468990, 459317, 454718, 452407, 449429, 448313, 446984, 4442…
$ UTM_Y     <int> 4482106, 4478341, 4477023, 4474944, 4472464, 4470879, 447034…
$ COD_SUCA  <chr> "005A0549", "14801866", "04910029", "049A0172", "07908410", …
$ geometry  <POINT [m]> POINT (977597.5 4497316), POINT (968176.9 4492889), PO…


Muestra de datos:

In [31]:
data_04 |> slice_head(n = 5)

LINEA         MUNICIPIO         CMUN CONEXION                        
1 C-2, C-7, C-8 Alcalá de Henares 005  Aparcamiento                    
2 C-2, C-7, C-8 Torrejón de Ardoz 148  Aparcamiento                    
3 C-2, C-7, C-8 Coslada           049  Aparcamiento                    
4 C-2, C-7, C-8 Coslada           049  Aparcamiento, conexión L-7 metro
5 C-2, C-7, C-8 Madrid            079  Aparcamiento, conexión L-9 metro
  ETIQUETA                UTM_X  UTM_Y   COD_SUCA geometry                
1 Alcalá de Henares       468990 4482106 005A0549 POINT (977597.5 4497316)
2 Torrejón de Ardoz       459317 4478341 14801866 POINT (968176.9 4492889)
3 San Fernando de Henares 454718 4477023 04910029 POINT (963665.9 4491257)
4 Coslada                 452407 4474944 049A0172 POINT (961496.3 4489020)
5 Vicálvaro               449429 4472464 07908410 POINT (958686.4 4486336)

### 5. Helisuperficies

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 


Data load using the {tcltk} package. Ucomment the line if not using this option

In [ ]:
# file_data_05 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>

NOTA: se omiten las comprobaciones de existencia de archivo para mejor compresión del código

In [32]:
iFile_05 <- "Transportes y comunicaciones_ Helisuperficies/"
file_data_05 <- paste0(iPath, iFile_05)


#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [37]:
data_05 <- st_read(file_data_05,
                            options = "ENCODING=latin1", 
                            quiet = FALSE)


options:        ENCODING=latin1 
Reading layer `helisuper' from data source 
  `/Users/emilio.lcano/academico/gh_repos/__transferencia/citizenlab/CitizenLab-Research-and-Development/casos_urjc/notebooks/II_data_processing/18_infraestructuras/Data/Input/Transportes y comunicaciones_ Helisuperficies' 
  using driver `ESRI Shapefile'
Simple feature collection with 87 features and 8 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 373754 ymin: 4434452 xmax: 490376 ymax: 4550969
Projected CRS: ETRS89 / UTM zone 30N


Estructura de  los datos:

In [38]:
glimpse(data_05)

Rows: 87
Columns: 9
$ CMUN      <chr> "0133", "0604", "0528", "1570", "0552", "0259", "0915", "091…
$ MUNICIPIO <chr> "Aranjuez", "Fuentidueña de Tajo", "Chinchón", "Valdelaguna"…
$ ETIQUETA  <chr> "Helipuerto Hospital del Tajo", "Helipuerto Eventual de Fuen…
$ USO       <chr> "Sanitarias/ Emergencias", "Emergencias Sanitarias", "Emerge…
$ GESTOR    <chr> "Hospital del Tajo (Aranjuez)", "Ayuntamiento de Fuentidueña…
$ UTM_X     <dbl> 447862, 485991, 463295, 467472, 490106, 490376, 461962, 4591…
$ UTM_Y     <dbl> 4434452, 4441209, 4444189, 4445339, 4449213, 4453110, 445350…
$ CLAVE     <chr> "HPUR", "HUEOEE", "HUEOEE", "ACUPUR", "HUEOEE", "HUEOEE", "H…
$ geometry  <POINT [m]> POINT (447862 4434452), POINT (485991 4441209), POINT …


Muestra de datos:

In [39]:
data_05 |> slice_head(n = 5)

CMUN MUNICIPIO           ETIQUETA                                   
1 0133 Aranjuez            Helipuerto Hospital del Tajo               
2 0604 Fuentidueña de Tajo Helipuerto Eventual de Fuentidueña del Tajo
3 0528 Chinchón            Helipuerto Eventual de Chinchón            
4 1570 Valdelaguna         Campo de Ultraligeros de Valdelaguna       
5 0552 Estremera           Helipuerto Eventual de Estremera           
  USO                         GESTOR                              UTM_X 
1 Sanitarias/ Emergencias     Hospital del Tajo (Aranjuez)        447862
2 Emergencias Sanitarias      Ayuntamiento de Fuentidueña de Tajo 485991
3 Emergencias Sanitarias      Ayuntamiento de Chinchón            463295
4 Aviación Deportiva/ Escuela Carina Delgado Ramos.               467472
5 Emergencias Sanitarias      Ayuntamiento de Estremera           490106
  UTM_Y   CLAVE  geometry              
1 4434452 HPUR   POINT (447862 4434452)
2 4441209 HUEOEE POINT (485991 4441209)
3 4444189 HUEOEE POINT (463295 4444189)
4 4445339 ACUPUR POINT (467472 4445339)
5 4449213 HUEOEE POINT (490106 4449213)

### 6. Centros de Atención Integral a Drogodependiente

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 


Data load using the {tcltk} package. Ucomment the line if not using this option

In [ ]:
# file_data_06 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>

NOTA: se omiten las comprobaciones de existencia de archivo para mejor compresión del código

In [40]:
iFile_06 <- "Salud_ Centros sanitarios. Centros de Atención Integral a Drogodependientes/drogodependencia.shp"
file_data_06 <- paste0(iPath, iFile_06)


#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [41]:
data_06 <- st_read(file_data_06,
                            options = "ENCODING=latin1", 
                            quiet = FALSE)


options:        ENCODING=latin1 
Reading layer `drogodependencia' from data source 
  `/Users/emilio.lcano/academico/gh_repos/__transferencia/citizenlab/CitizenLab-Research-and-Development/casos_urjc/notebooks/II_data_processing/18_infraestructuras/Data/Input/Salud_ Centros sanitarios. Centros de Atención Integral a Drogodependientes/drogodependencia.shp' 
  using driver `ESRI Shapefile'
Simple feature collection with 35 features and 9 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 380860 ymin: 4431637 xmax: 468812 ymax: 4501031
Projected CRS: ETRS89 / UTM zone 30N


Estructura de  los datos:

In [42]:
glimpse(data_06)

Rows: 35
Columns: 10
$ CODMUN    <chr> "005", "006", "007", "013", "014", "047", "045", "049", "079…
$ NOMBRE    <chr> "Centro de Atención Integral a Drogodependientes Alcalá de H…
$ DIRECCION <chr> "C/ Santiago, 13 - planta", "C/ Libertad, 6 - 2º", "Paseo de…
$ UTM_X     <dbl> 468812, 446255, 429345, 449413, 463258, 414669, 434697, 4531…
$ UTM_Y     <dbl> 4481428, 4487944, 4466846, 4431637, 4460592, 4498050, 450103…
$ MUNICIPIO <chr> "Alcalá de Henares", "Alcobendas", "Alcorcón", "Aranjuez", "…
$ URL       <chr> "http://centrossanitarios.sanidadmadrid.org/Otros/Detalle.as…
$ CODIGO2   <int> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 1…
$ COD_SUCA  <chr> "005A0696", "006A0179", "00701867", "013A0208", "014A0253", …
$ geometry  <POINT [m]> POINT (468812 4481428), POINT (446255 4487944), POINT …


Muestra de datos:

In [43]:
data_06 |> slice_head(n = 5)

CODMUN NOMBRE                                                           
1 005    Centro de Atención Integral a Drogodependientes Alcalá de Henares
2 006    Centro de Atención Integral a Drogodependientes Alcobendas       
3 007    Centro de Atención Integral a Drogodependientes Alcorcón         
4 013    Centro de Atención Integral a Drogodependientes Aranjuez         
5 014    Centro de Atención Integral a Drogodependientes Arganda del Rey  
  DIRECCION                 UTM_X  UTM_Y   MUNICIPIO        
1 C/ Santiago, 13 - planta  468812 4481428 Alcalá de Henares
2 C/ Libertad, 6 - 2º       446255 4487944 Alcobendas       
3 Paseo de Extremadura, 19  429345 4466846 Alcorcón         
4 C/ Primero de Mayo, 1     449413 4431637 Aranjuez         
5 C/ Camino del Molino, s/n 463258 4460592 Arganda del Rey  
  URL                                                                  CODIGO2
1 http://centrossanitarios.sanidadmadrid.org/Otros/Detalle.aspx?ID=576 1      
2 http://centrossanitarios.sanidadmadrid.org/Otros/Detalle.aspx?ID=577 2      
3 http://centrossanitarios.sanidadmadrid.org/Otros/Detalle.aspx?ID=578 3      
4 http://centrossanitarios.sanidadmadrid.org/Otros/Detalle.aspx?ID=579 4      
5 http://centrossanitarios.sanidadmadrid.org/Otros/Detalle.aspx?ID=580 5      
  COD_SUCA geometry              
1 005A0696 POINT (468812 4481428)
2 006A0179 POINT (446255 4487944)
3 00701867 POINT (429345 4466846)
4 013A0208 POINT (449413 4431637)
5 014A0253 POINT (463258 4460592)

### 7. Centros de especialidades

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 


Data load using the {tcltk} package. Ucomment the line if not using this option

In [ ]:
# file_data_07 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>

NOTA: se omiten las comprobaciones de existencia de archivo para mejor compresión del código

In [44]:
iFile_07 <- "Salud_ Centros sanitarios. Centros de especialidades/centroespecialidades.shp"
file_data_07 <- paste0(iPath, iFile_07)


#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [45]:
data_07 <- st_read(file_data_07,
                            options = "ENCODING=latin1", 
                            quiet = FALSE)


options:        ENCODING=latin1 
Reading layer `centroespecialidades' from data source 
  `/Users/emilio.lcano/academico/gh_repos/__transferencia/citizenlab/CitizenLab-Research-and-Development/casos_urjc/notebooks/II_data_processing/18_infraestructuras/Data/Input/Salud_ Centros sanitarios. Centros de especialidades/centroespecialidades.shp' 
  using driver `ESRI Shapefile'
Simple feature collection with 28 features and 15 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 402704.8 ymin: 4454857 xmax: 468379.8 ymax: 4501383
Projected CRS: ETRS89 / UTM zone 30N


Estructura de  los datos:

In [46]:
glimpse(data_07)

Rows: 28
Columns: 16
$ ETIQUETA  <chr> "C.E. Moratalaz", "C.E. Peña Prieta - Hermanos Sangró", "C.E…
$ TIPOCEN2  <chr> "Centro de Especialidades", "Centro de Especialidades", "Cen…
$ TIPOVIAL  <chr> "Calle", "Avenida", "Calle", "Avenida", "Calle", "Avenida", …
$ NOMVIA    <chr> "Hacienda de Pavones", "Peña Prieta", "Sierra de Alquife", "…
$ NUMERO    <chr> "348", "4", "8", "285", "45", "s/n", "11", "30", "89", "13",…
$ MUNICIPIO <chr> "Madrid", "Madrid", "Madrid", "Madrid", "Madrid", "Torrejón …
$ CODMUN    <chr> "0796", "0796", "0796", "0796", "0796", "1489", "0053", "079…
$ CODPOS    <chr> "28030", "28038", "28018", "28038", "28028", "28850", "28806…
$ CODAREA   <chr> "01", "01", "01", "01", "02", "03", "03", "04", "04", "05", …
$ CODDISTR  <chr> "0102", "0104", "0104", "0104", "0202", "0302", "0301", "040…
$ CODZBS    <chr> "010203", "010402", "010406", "010410", "020201", "030202", …
$ CODVIA    <chr> "07904590", "07904676", "07909818", "07900130", "07901736", …
$ UTM_X     <dbl> 4

Muestra de datos:

In [47]:
data_07 |> slice_head(n = 5)

ETIQUETA                           TIPOCEN2                 TIPOVIAL
1 C.E. Moratalaz                     Centro de Especialidades Calle   
2 C.E. Peña Prieta - Hermanos Sangró Centro de Especialidades Avenida 
3 C.E. Vicente Soldevilla            Centro de Especialidades Calle   
4 C.E. Federica Montseny             Centro de Especialidades Avenida 
5 C.E. Hermanos García Noblejas      Centro de Especialidades Calle   
  NOMVIA              NUMERO MUNICIPIO CODMUN CODPOS CODAREA CODDISTR CODZBS
1 Hacienda de Pavones 348    Madrid    0796   28030  01      0102     010203
2 Peña Prieta         4      Madrid    0796   28038  01      0104     010402
3 Sierra de Alquife   8      Madrid    0796   28018  01      0104     010406
4 Albufera            285    Madrid    0796   28038  01      0104     010410
5 Doctor Esquerdo     45     Madrid    0796   28028  02      0202     020201
  CODVIA   UTM_X  UTM_Y   CODIGO2 geometry                
1 07904590 446223 4472348 1       POINT (446222.8 4472348)
2 07904676 443279 4472215 2       POINT (443278.6 4472215)
3 07909818 443380 4471141 3       POINT (443379.6 4471141)
4 07900130 445704 4470967 4       POINT (445703.5 4470967)
5 07901736 443300 4474985 5       POINT (443299.6 4474985)

### 8. Centros de salud

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 


Data load using the {tcltk} package. Ucomment the line if not using this option

In [ ]:
# file_data_08 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>

NOTA: se omiten las comprobaciones de existencia de archivo para mejor compresión del código

In [48]:
iFile_08 <- "Salud_ Centros sanitarios. Centros de salud/centrosalud.shp"
file_data_08 <- paste0(iPath, iFile_08)


#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [49]:
data_08 <- st_read(file_data_08,
                            options = "ENCODING=latin1", 
                            quiet = FALSE)


options:        ENCODING=latin1 
Reading layer `centrosalud' from data source 
  `/Users/emilio.lcano/academico/gh_repos/__transferencia/citizenlab/CitizenLab-Research-and-Development/casos_urjc/notebooks/II_data_processing/18_infraestructuras/Data/Input/Salud_ Centros sanitarios. Centros de salud/centrosalud.shp' 
  using driver `ESRI Shapefile'
Simple feature collection with 267 features and 14 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 377838.2 ymin: 4430802 xmax: 476562 ymax: 4538179
Projected CRS: ETRS89 / UTM zone 30N


Estructura de  los datos:

In [50]:
glimpse(data_08)

Rows: 267
Columns: 15
$ TIPOCEN2   <chr> "Centro de Salud Carmen Calzado", "Centro de Salud Juan de …
$ TIPOVIAL   <chr> "Calle", "Avenida", "Calle", "Calle", "Calle", "Calle", "Av…
$ NOMVIA     <chr> "Carmen Calzado", "Juan de Austria", "Arturo Soria", "Luis …
$ NUMERO     <chr> "14", "19", "12", "16", "1", "1", "23", "12", "61", "12", "…
$ MUNICIPIO  <chr> "Alcalá de Henares", "Alcalá de Henares", "Alcalá de Henare…
$ CMUN       <chr> "005", "005", "005", "005", "005", "005", "005", "005", "00…
$ CODVIA     <chr> "005A0150", "005A0370", "00500730", "005A0424", "005A0441",…
$ UTM_X      <dbl> 468979, 470297, 466617, 468796, 469712, 469685, 468549, 468…
$ UTM_Y      <dbl> 4481282, 4481952, 4481245, 4480551, 4481336, 4482126, 44844…
$ CODIGO2    <int> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, …
$ SUCA_CODVI <chr> "005A0150", "005A0370", "00500730", "005A0424", "005A0441",…
$ CODZBS     <chr> "030101", "030107", "030109", "030104", "030105", "030108",…
$ CODDISTR   <chr>

Muestra de datos:

In [51]:
data_08 |> slice_head(n = 5)

TIPOCEN2                        TIPOVIAL NOMVIA          NUMERO
1 Centro de Salud Carmen Calzado  Calle    Carmen Calzado  14    
2 Centro de Salud Juan de Austria Avenida  Juan de Austria 19    
3 Centro de Salud La Garena       Calle    Arturo Soria    12    
4 Centro de Salud Luis Vives      Calle    Luis Vives      16    
5 Centro de Salud Manuel Merino   Calle    Manuel Merino   1     
  MUNICIPIO         CMUN CODVIA   UTM_X  UTM_Y   CODIGO2 SUCA_CODVI CODZBS
1 Alcalá de Henares 005  005A0150 468979 4481282 1       005A0150   030101
2 Alcalá de Henares 005  005A0370 470297 4481952 2       005A0370   030107
3 Alcalá de Henares 005  00500730 466617 4481245 3       00500730   030109
4 Alcalá de Henares 005  005A0424 468796 4480551 4       005A0424   030104
5 Alcalá de Henares 005  005A0441 469712 4481336 5       005A0441   030105
  CODDISTR CODAREA geometry              
1 0301     03      POINT (468979 4481282)
2 0301     03      POINT (470297 4481952)
3 0301     03      POINT (466617 4481245)
4 0301     03      POINT (468796 4480551)
5 0301     03      POINT (469712 4481336)

### 9. Consultorios de salud

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 


Data load using the {tcltk} package. Ucomment the line if not using this option

In [ ]:
# file_data_09 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>

NOTA: se omiten las comprobaciones de existencia de archivo para mejor compresión del código

In [52]:
iFile_09 <- "Salud_ Centros sanitarios. Consultorios de salud/consultoriosalud.shp"
file_data_09 <- paste0(iPath, iFile_09)


#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [53]:
data_09 <- st_read(file_data_09,
                            options = "ENCODING=latin1", 
                            quiet = FALSE)


options:        ENCODING=latin1 
Reading layer `consultoriosalud' from data source 
  `/Users/emilio.lcano/academico/gh_repos/__transferencia/citizenlab/CitizenLab-Research-and-Development/casos_urjc/notebooks/II_data_processing/18_infraestructuras/Data/Input/Salud_ Centros sanitarios. Consultorios de salud/consultoriosalud.shp' 
  using driver `ESRI Shapefile'
Simple feature collection with 156 features and 10 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 373181 ymin: 4435469 xmax: 490638 ymax: 4553677
Projected CRS: ETRS89 / UTM zone 30N


Estructura de  los datos:

In [54]:
glimpse(data_09)

Rows: 156
Columns: 11
$ ETIQUETA  <chr> "Consultorio de La Acebeda", "Consultorio de Ajalvir", "Cons…
$ TIPOVIA   <chr> "Plaza", "Avda", "Plaza", "Calle", "Calle", "Avda", "Plaza",…
$ NOMVIA    <chr> "San Miguel", "Aguas", "Santa Marina", "Federico García Lorc…
$ NUMERO    <chr> "1", "2", "17", "31", "17", "3", "1", "15", "2", "11", "s/n"…
$ MUNICIPIO <chr> "La Acebeda", "Ajalvir", "Alameda del Valle", "El Álamo", "A…
$ CODMUNI   <chr> "0014", "0029", "0035", "0040", "0088", "0091", "0105", "011…
$ CODVIA    <chr> "00100028", "00200141", "003A0030", "00400218", "008A0088", …
$ UTM_X     <int> 447522, 459130, 428958, 415892, 398120, 450987, 413034, 4844…
$ UTM_Y     <dbl> 4548589, 4487429, 4530000, 4454069, 4464011, 4496981, 450118…
$ COD2      <int> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 1…
$ geometry  <POINT [m]> POINT (447522 4548589), POINT (459130 4487429), POINT …


Muestra de datos:

In [55]:
data_09 |> slice_head(n = 5)

ETIQUETA                         TIPOVIA NOMVIA                NUMERO
1 Consultorio de La Acebeda        Plaza   San Miguel            1     
2 Consultorio de Ajalvir           Avda    Aguas                 2     
3 Consultorio de Alameda del Valle Plaza   Santa Marina          17    
4 Consultorio de El Álamo          Calle   Federico García Lorca 31    
5 Consultorio de Aldea del Fresno  Calle   San Rafael            17    
  MUNICIPIO         CODMUNI CODVIA   UTM_X  UTM_Y   COD2 geometry              
1 La Acebeda        0014    00100028 447522 4548589 1    POINT (447522 4548589)
2 Ajalvir           0029    00200141 459130 4487429 2    POINT (459130 4487429)
3 Alameda del Valle 0035    003A0030 428958 4530000 3    POINT (428958 4530000)
4 El Álamo          0040    00400218 415892 4454069 4    POINT (415892 4454069)
5 Aldea del Fresno  0088    008A0088 398120 4464011 5    POINT (398120 4464011)

### 10. Otros Centros

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 


Data load using the {tcltk} package. Ucomment the line if not using this option

In [58]:
# file_data_10 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>

NOTA: se omiten las comprobaciones de existencia de archivo para mejor compresión del código

In [59]:
iFile_10 <- "Salud_ Centros sanitarios. Otros centros/saludotros.shp"
file_data_10 <- paste0(iPath, iFile_10)


#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [60]:
data_10 <- st_read(file_data_10,
                            options = "ENCODING=latin1", 
                            quiet = FALSE)


options:        ENCODING=latin1 
Reading layer `saludotros' from data source 
  `/Users/emilio.lcano/academico/gh_repos/__transferencia/citizenlab/CitizenLab-Research-and-Development/casos_urjc/notebooks/II_data_processing/18_infraestructuras/Data/Input/Salud_ Centros sanitarios. Otros centros/saludotros.shp' 
  using driver `ESRI Shapefile'
Simple feature collection with 29 features and 7 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 434198.7 ymin: 4463663 xmax: 468813 ymax: 4489946
Projected CRS: ETRS89 / UTM zone 30N


Estructura de  los datos:

In [61]:
glimpse(data_10)

Rows: 29
Columns: 8
$ CODMUN    <chr> "079", "079", "079", "079", "079", "079", "134", "079", "079…
$ NOMBRE    <chr> "Instituto Provincial de Rehabilitación", "Instituto Provinc…
$ DIRECCION <chr> "Calle Francisco Silvela, 40", "Calle General Arrando, 17", …
$ CODIGO2   <int> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 1…
$ UTM_X     <dbl> 442944, 441216, 441846, 441859, 441873, 441821, 448297, 4477…
$ UTM_Y     <dbl> 4476050, 4475926, 4480831, 4480830, 4480830, 4476418, 448994…
$ COD_SUCA  <chr> "07902340", "07902499", "07901213", "07901213", "07901213", …
$ geometry  <POINT [m]> POINT (442944 4476050), POINT (441216 4475926), POINT …


Muestra de datos:

In [62]:
data_10 |> slice_head(n = 5)

CODMUN NOMBRE                                          
1 079    Instituto Provincial de Rehabilitación          
2 079    Instituto Provincial de Oftalmología            
3 079    Oficina Regional de Coordinación de Transplantes
4 079    Oficina Regional de Coordinación de Salud Mental
5 079    Oficina Regional de Coordinación Oncológica     
  DIRECCION                   CODIGO2 UTM_X  UTM_Y   COD_SUCA
1 Calle Francisco Silvela, 40 1       442944 4476050 07902340
2 Calle General Arrando, 17   2       441216 4475926 07902499
3 Paseo de la Castellana, 280 3       441846 4480831 07901213
4 Paseo de la Castellana, 280 4       441859 4480830 07901213
5 Paseo de la Castellana, 280 5       441873 4480830 07901213
  geometry                
1 POINT (442944 4476050)  
2 POINT (441216 4475926)  
3 POINT (441845.9 4480831)
4 POINT (441858.9 4480830)
5 POINT (441872.9 4480830)

### 11. Centros de salud mental

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 


Data load using the {tcltk} package. Ucomment the line if not using this option

In [ ]:
# file_data_11 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>

NOTA: se omiten las comprobaciones de existencia de archivo para mejor compresión del código

In [63]:
iFile_11 <- "Salud_ Centros sanitarios. Servicio de salud mental de distrito/saludmental.shp"
file_data_11 <- paste0(iPath, iFile_11)


#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [64]:
data_11 <- st_read(file_data_11,
                            options = "ENCODING=latin1", 
                            quiet = FALSE)


options:        ENCODING=latin1 
Reading layer `saludmental' from data source 
  `/Users/emilio.lcano/academico/gh_repos/__transferencia/citizenlab/CitizenLab-Research-and-Development/casos_urjc/notebooks/II_data_processing/18_infraestructuras/Data/Input/Salud_ Centros sanitarios. Servicio de salud mental de distrito/saludmental.shp' 
  using driver `ESRI Shapefile'
Simple feature collection with 53 features and 10 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 402720 ymin: 4434284 xmax: 468396 ymax: 4501363
Projected CRS: ETRS89 / UTM zone 30N


Estructura de  los datos:

In [65]:
glimpse(data_11)

Rows: 53
Columns: 11
$ NOMBRE     <chr> "CSM Villa Vallecas", "CSM Coslada", "CSM Salamanca", "CSM …
$ TIPOVIAL   <chr> "Calle", "Avenida", "Calle", "Calle", "Calle", "Avenida", "…
$ NOMVIA     <chr> "San Claudio", "España", "O'Donnell", "Aguileñas", "Octavio…
$ MUNICIPIO  <chr> "Madrid", "Coslada", "Madrid", "Madrid", "Alcalá de Henares…
$ NUMERO     <chr> "154", "s/n", "55", "1", "15", "4", "50", "20", "1", "11", …
$ CODMUN     <chr> "0796", "0493", "0796", "0796", "0053", "1489", "0066", "13…
$ CODIGO2    <int> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, …
$ SUCA_CDVIA <chr> "07910251", "049A0101", "07904389", "07900067", "00504560",…
$ UTM_X      <dbl> 445763, 453151, 443046, 441192, 468396, 461608, 446048, 446…
$ UTM_Y      <dbl> 4471085, 4474618, 4474762, 4479657, 4483323, 4478644, 44887…
$ geometry   <POINT [m]> POINT (445763 4471085), POINT (453151 4474618), POINT…


Muestra de datos:

In [66]:
data_11 |> slice_head(n = 5)

NOMBRE                        TIPOVIAL NOMVIA      MUNICIPIO         NUMERO
1 CSM Villa Vallecas            Calle    San Claudio Madrid            154   
2 CSM Coslada                   Avenida  España      Coslada           s/n   
3 CSM Salamanca                 Calle    O'Donnell   Madrid            55    
4 CSM Tetuán Infanto Juvenil    Calle    Aguileñas   Madrid            1     
5 CSM Alcalá  de Henarés I y II Calle    Octavio Paz Alcalá de Henares 15    
  CODMUN CODIGO2 SUCA_CDVIA UTM_X  UTM_Y   geometry              
1 0796   1       07910251   445763 4471085 POINT (445763 4471085)
2 0493   2       049A0101   453151 4474618 POINT (453151 4474618)
3 0796   3       07904389   443046 4474762 POINT (443046 4474762)
4 0796   4       07900067   441192 4479657 POINT (441192 4479657)
5 0053   5       00504560   468396 4483323 POINT (468396 4483323)

### 12. Hospitales

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 


Data load using the {tcltk} package. Ucomment the line if not using this option

In [ ]:
# file_data_12 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>

NOTA: se omiten las comprobaciones de existencia de archivo para mejor compresión del código

In [127]:
iFile_12 <- "hospitales_poi.csv"
file_data_12 <- paste0(iPath, iFile_12)


#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [128]:
data_12 <- read_csv(file_data_12)


Rows: 87 Columns: 67
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (31): Nombre Centro, Dirección, Teléfono, Municipio, Provincia, CCAA, Có...
dbl (35): CCN, CODCNH, Cód. Municipio, Cód. Provincia, Cód. CCAA, Código Pos...
lgl  (1): CIERREFECH

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Estructura de  los datos:

In [129]:
glimpse(data_12)

Rows: 87
Columns: 67
$ CCN                     <dbl> 1328020932, 1328000034, 1328000070, 1328000066…
$ CODCNH                  <dbl> 281371, 280745, 281124, 281071, 281292, 281281…
$ `Nombre Centro`         <chr> "HOSPITAL HM VALLES", "HOSPITAL UNIVERSITARIO …
$ Dirección               <chr> "CALLE SANTIAGO 14", "CARRETERA DE MECO S/N", …
$ Teléfono                <chr> "918883262", "918878100", "917810682", "916219…
$ `Cód. Municipio`        <dbl> 280053, 280053, 280072, 280072, 280133, 280148…
$ Municipio               <chr> "Alcalá de Henares", "Alcalá de Henares", "Alc…
$ `Cód. Provincia`        <dbl> 28, 28, 28, 28, 28, 28, 28, 28, 28, 28, 28, 28…
$ Provincia               <chr> "MADRID", "MADRID", "MADRID", "MADRID", "MADRI…
$ `Cód. CCAA`             <dbl> 13, 13, 13, 13, 13, 13, 13, 13, 13, 13, 13, 13…
$ CCAA                    <chr> "MADRID", "MADRID", "MADRID", "MADRID", "MADRI…
$ `Código Postal`         <dbl> 28801, 28805, 28922, 28922, 28300, 28500, 2866…
$ CAMAS            

Muestra de datos:

In [124]:
data_12 |> slice_head(n = 5)

CCN        CODCNH Nombre Centro                              
1 1328020932 281371 HOSPITAL HM VALLES                         
2 1328000034 280745 HOSPITAL UNIVERSITARIO PRINCIPE DE ASTURIAS
3 1328000070 281124 HOSPITAL QUIRONSALUD SUR                   
4 1328000066 281071 HOSPITAL UNIVERSITARIO FUNDACION ALCORCON  
5 1328000084 281292 HOSPITAL DEL TAJO S.A.                     
  Dirección                    Teléfono  Cód. Municipio Municipio        
1 CALLE SANTIAGO 14            918883262 280053         Alcalá de Henares
2 CARRETERA DE MECO S/N        918878100 280053         Alcalá de Henares
3 CALLE DE ESTAMBUL 30         917810682 280072         Alcorcón         
4 CALLE BUDAPEST 1             916219400 280072         Alcorcón         
5 AVENIDA AMAZONAS CENTRAL S/N 918014100 280133         Aranjuez         
  Cód. Provincia Provincia Cód. CCAA ⋯ SPECT PET MAMOS DO DIAL X        
1 28             MADRID    13        ⋯ 0     0   1     1   0   -3.367372
2 28             MADRID    13        ⋯ 0     0   2     1  28   -3.347602
3 28             MADRID    13        ⋯ 0     0   1     1   0   -3.849035
4 28             MADRID    13        ⋯ 0     0   2     1  28   -3.837394
5 28             MADRID    13        ⋯ 0     0   1     0  13   -3.610795
  Y        CalidadGeocodificacion id_area nombre_area
1 40.48302 PointAddress           03      Este       
2 40.51025 Manual                 03      Este       
3 40.34248 StreetAddress          08      Sur-Oeste I
4 40.34887 Manual                 08      Sur-Oeste I
5 40.05726 Manual                 11      Sur Ii

## ETL Processes

### Import data from: CSV, Excel, Tab, JSON, SQL, and Parquet files

Se han importado en el apartado Data Load anterior:

* Localizaciones de POIS publicados en el portal de la Comunidad de Madrid
* Localización de hospitales del catálogo nacional de hospitales

Incluir apartados si procede para: Extracción de datos (select, filter), Transformación de datos, (mutate, joins, ...). Si es necesario tratar datos perdidos, indicarlo también en NB 09.2

### Data transformation

Para cada data frame de infraestructuras se realizan las siguientes transformaciones:

* Transformar sistema de referencia CRS a 4326 (WGS 84)
* Añadir las coordenadas como columnas normales
* Establecer variables `grupo` y `tipo`
* Si hay información adicional en el archivo, añadirla a columna `info`
* Asignar a la columna `nombre` la columna más adecuada según lo importado
* Seleccionar columnas que se usarán en todos los data frames a unir
* Eliminar la columna de geometría para que se pueda exportar a csv

Para los hospitales igual pero sin añadir coordenadas que ya las tenía ni eliminar la geometría porque no es objeto sf.

Finalmente se unen todos los data.frames en uno


In [84]:
tdata_01 <- data_01 |>
  bind_cols(st_coordinates(data_01 |> st_transform(4326))) |> 
  mutate(grupo = "Transporte",
         tipo = "Intercambiadores",
         info = "") |> 
  select(grupo,
         tipo, 
         nombre = ETIQUETA, 
         CODMUN = CMUN, 
         DIRECCION, 
         X, Y,
         info) |> 
  st_drop_geometry()


In [85]:
tdata_01 |> slice_head(n = 5)

grupo      tipo             nombre                                     
1 Transporte Intercambiadores Grandes Intercambiadores Plaza Elíptica    
2 Transporte Intercambiadores Grandes Intercambiadores Moncloa           
3 Transporte Intercambiadores Grandes Intercambiadores Príncipe Pío      
4 Transporte Intercambiadores Grandes Intercambiadores Plaza de Castilla 
5 Transporte Intercambiadores Grandes Intercambiadores Avenida de América
  CODMUN DIRECCION                                X         Y        info
1 079    Plaza Elíptica s/n                       -3.716577 40.38540     
2 079    Calle Princesa, 89                       -3.719508 40.43474     
3 079    Estación Príncipe Pío                    -3.719560 40.42080     
4 079    Calle San Aquilino y Avenida de Asturias -3.689044 40.46719     
5 079    Avenida de América 9 A                   -3.676731 40.43797

In [88]:
tdata_02 <- data_02 |>
  bind_cols(st_coordinates(data_02 |> st_transform(4326))) |> 
    mutate(grupo = "Transporte",
         tipo = "Bocas de metro",
         ESTACION = paste(ESTACION, SALIDA, sep = " - "),
         info = paste0("Líneas: ", LINEAS )) |> 
  select(grupo,
         tipo, 
         nombre = ESTACION, 
         CODMUN = COD_MUNI, 
         DIRECCION, 
         X, Y,
         info) |> 
  st_drop_geometry()


In [89]:
tdata_02 |> slice_head(n = 5)

grupo      tipo           nombre                                     CODMUN
1 Transporte Bocas de metro Plaza de Castilla - Castellana             079   
2 Transporte Bocas de metro Plaza de Castilla - Plaza de Castilla      079   
3 Transporte Bocas de metro Plaza de Castilla - Pº Castellana, impares 079   
4 Transporte Bocas de metro Valdeacederas - Capitán Blanco Argibay     079   
5 Transporte Bocas de metro Valdeacederas - Aníbal                     079   
  DIRECCION                               X         Y        info            
1 Frente depósito C. Isabel II            -3.688866 40.46549 Líneas: 1, 10, 9
2 Paseo de la Castellana, bulevar central -3.689186 40.46684 Líneas: 1, 10, 9
3 Paseo Castellana, 189 (junto a torres)  -3.689532 40.46718 Líneas: 1, 10, 9
4 Calle Bravo Murillo, 324                -3.695787 40.46409 Líneas: 1       
5 Calle Bravo Murillo, 350                -3.694037 40.46478 Líneas: 1

In [91]:
tdata_03 <- data_03 |>
  bind_cols(st_coordinates(data_03 |> st_transform(4326))) |> 
  mutate(grupo = "Transporte",
         tipo = "Aeropuertos",
         info = "") |> 
  select(grupo,
         tipo, 
         nombre = BUSCA, 
         CODMUN = CMUN, 
         DIRECCION = DIRECCIÓN, 
         X, Y,
         info) |>   st_drop_geometry()


In [92]:
tdata_03 |> slice_head(n = 5)

grupo      tipo        nombre                                  CODMUN
1 Transporte Aeropuertos Aeropuerto Adolfo Suárez-Madrid Barajas 079   
2 Transporte Aeropuertos Aeropuerto Madrid-Cuatro Vientos        065   
3 Transporte Aeropuertos Aeropuerto Madrid-Torrejón de Ardoz     148   
  DIRECCION                       X         Y        info
1 Avda. Hispanidad, s/n           -3.571989 40.46012     
2 Ctra. Barrio de la Fortuna, s/n -3.774744 40.37036     
3 Ctra. de la Base, s/n           -3.446897 40.47660

In [98]:
tdata_04 <- data_04 |>
  bind_cols(st_coordinates(data_04 |> st_transform(4326))) |> 
  mutate(grupo = "Transporte",
         tipo = "Estaciones de Cercanías",
         DIRECCION = CMUN,
         info = paste0("Líneas: ", LINEA)) |> 
  select(grupo,
         tipo, 
         nombre = ETIQUETA, 
         CODMUN = CMUN, 
         DIRECCION, 
         X, Y,
         info) |> 
  st_drop_geometry()


In [99]:
tdata_04 |> slice_head(n = 5)

grupo      tipo                    nombre                  CODMUN DIRECCION
1 Transporte Estaciones de Cercanías Alcalá de Henares       005    005      
2 Transporte Estaciones de Cercanías Torrejón de Ardoz       148    148      
3 Transporte Estaciones de Cercanías San Fernando de Henares 049    049      
4 Transporte Estaciones de Cercanías Coslada                 049    049      
5 Transporte Estaciones de Cercanías Vicálvaro               079    079      
  X         Y        info                 
1 -3.365905 40.48908 Líneas: C-2, C-7, C-8
2 -3.479804 40.45474 Líneas: C-2, C-7, C-8
3 -3.533951 40.44263 Líneas: C-2, C-7, C-8
4 -3.561040 40.42377 Líneas: C-2, C-7, C-8
5 -3.595947 40.40125 Líneas: C-2, C-7, C-8

In [100]:
tdata_05 <- data_05 |>
  bind_cols(st_coordinates(data_05 |> st_transform(4326))) |> 
  mutate(grupo = "Transporte",
         tipo = "Helisuperficies",
         DIRECCION = MUNICIPIO,
         info = paste0("Uso: ", USO, "; Gestor: ", GESTOR)) |> 
  select(grupo,
         tipo, 
         nombre = ETIQUETA, 
         CODMUN = CMUN, 
         DIRECCION, 
         X, Y,
         info) |> 
  st_drop_geometry()


In [101]:
tdata_05 |> slice_head(n = 5)

grupo      tipo            nombre                                      CODMUN
1 Transporte Helisuperficies Helipuerto Hospital del Tajo                0133  
2 Transporte Helisuperficies Helipuerto Eventual de Fuentidueña del Tajo 0604  
3 Transporte Helisuperficies Helipuerto Eventual de Chinchón             0528  
4 Transporte Helisuperficies Campo de Ultraligeros de Valdelaguna        1570  
5 Transporte Helisuperficies Helipuerto Eventual de Estremera            0552  
  DIRECCION           X         Y       
1 Aranjuez            -3.611325 40.05871
2 Fuentidueña de Tajo -3.164408 40.12108
3 Chinchón            -3.430930 40.14724
4 Valdelaguna         -3.381950 40.15778
5 Estremera           -3.116238 40.19325
  info                                                                    
1 Uso: Sanitarias/ Emergencias; Gestor: Hospital del Tajo (Aranjuez)      
2 Uso: Emergencias Sanitarias; Gestor: Ayuntamiento de Fuentidueña de Tajo
3 Uso: Emergencias Sanitarias; Gestor: Ayuntamiento de Chinchón           
4 Uso: Aviación Deportiva/ Escuela; Gestor: Carina Delgado Ramos.         
5 Uso: Emergencias Sanitarias; Gestor: Ayuntamiento de Estremera

In [102]:
tdata_06 <- data_06 |>
  bind_cols(st_coordinates(data_06 |> st_transform(4326))) |> 
  mutate(tipo = "Centros de Atención a Drogodependientes",
         grupo = "Salud",
         info = "") |> 
  select(grupo,
         tipo, 
         nombre = NOMBRE, 
         CODMUN = CODMUN, 
         DIRECCION, 
         X, Y,
         info) |> 
  st_drop_geometry()


In [103]:
tdata_06 |> slice_head(n = 5)

grupo tipo                                   
1 Salud Centros de Atención a Drogodependientes
2 Salud Centros de Atención a Drogodependientes
3 Salud Centros de Atención a Drogodependientes
4 Salud Centros de Atención a Drogodependientes
5 Salud Centros de Atención a Drogodependientes
  nombre                                                            CODMUN
1 Centro de Atención Integral a Drogodependientes Alcalá de Henares 005   
2 Centro de Atención Integral a Drogodependientes Alcobendas        006   
3 Centro de Atención Integral a Drogodependientes Alcorcón          007   
4 Centro de Atención Integral a Drogodependientes Aranjuez          013   
5 Centro de Atención Integral a Drogodependientes Arganda del Rey   014   
  DIRECCION                 X         Y        info
1 C/ Santiago, 13 - planta  -3.367977 40.48296     
2 C/ Libertad, 6 - 2º       -3.634660 40.54050     
3 Paseo de Extremadura, 19  -3.831981 40.34918     
4 C/ Primero de Mayo, 1     -3.592921 40.03344     
5 C/ Camino del Molino, s/n -3.432303 40.29502

In [104]:
tdata_07 <- data_07 |>
  bind_cols(st_coordinates(data_07 |> st_transform(4326))) |> 
  mutate(tipo = "Centros de Especialidades",
         grupo = "Salud",
         info = paste0("CODZBS: ", CODZBS),
         DIRECCION = paste0(TIPOVIAL, NOMVIA, NUMERO)) |> 
  select(grupo,
         tipo, 
         nombre = ETIQUETA, 
         CODMUN = CODMUN, 
         DIRECCION, 
         X, Y,
         info) |> 
  st_drop_geometry()


In [105]:
tdata_07 |> slice_head(n = 5)

grupo tipo                      nombre                             CODMUN
1 Salud Centros de Especialidades C.E. Moratalaz                     0796  
2 Salud Centros de Especialidades C.E. Peña Prieta - Hermanos Sangró 0796  
3 Salud Centros de Especialidades C.E. Vicente Soldevilla            0796  
4 Salud Centros de Especialidades C.E. Federica Montseny             0796  
5 Salud Centros de Especialidades C.E. Hermanos García Noblejas      0796  
  DIRECCION                   X         Y        info          
1 CalleHacienda de Pavones348 -3.633718 40.40000 CODZBS: 010203
2 AvenidaPeña Prieta4         -3.668399 40.39861 CODZBS: 010402
3 CalleSierra de Alquife8     -3.667114 40.38895 CODZBS: 010406
4 AvenidaAlbufera285          -3.639719 40.38753 CODZBS: 010410
5 CalleDoctor Esquerdo45      -3.668399 40.42357 CODZBS: 020201

In [106]:
tdata_08 <- data_08 |>
  bind_cols(st_coordinates(data_08 |> st_transform(4326))) |> 
  mutate(tipo = "Centros de Salud",
         grupo = "Salud",
         info = paste0("CODZBS: ", CODZBS),
         DIRECCION = paste0(TIPOVIAL, NOMVIA, NUMERO)) |> 
  select(grupo,
         tipo, 
         nombre = TIPOCEN2, 
         CODMUN = CMUN, 
         DIRECCION, 
         X, Y,
         info) |> 
  st_drop_geometry()


In [107]:
tdata_08 |> slice_head(n = 5)

grupo tipo             nombre                          CODMUN
1 Salud Centros de Salud Centro de Salud Carmen Calzado  005   
2 Salud Centros de Salud Centro de Salud Juan de Austria 005   
3 Salud Centros de Salud Centro de Salud La Garena       005   
4 Salud Centros de Salud Centro de Salud Luis Vives      005   
5 Salud Centros de Salud Centro de Salud Manuel Merino   005   
  DIRECCION                X         Y        info          
1 CalleCarmen Calzado14    -3.365999 40.48165 CODZBS: 030101
2 AvenidaJuan de Austria19 -3.350480 40.48773 CODZBS: 030107
3 CalleArturo Soria12      -3.393864 40.48122 CODZBS: 030109
4 CalleLuis Vives16        -3.368122 40.47506 CODZBS: 030104
5 CalleManuel Merino1      -3.357354 40.48216 CODZBS: 030105

In [108]:
tdata_09 <- data_09 |>
  bind_cols(st_coordinates(data_09 |> st_transform(4326))) |> 
  mutate(tipo = "Consultorios de Salud",
         grupo = "Salud",
         info = "",
         DIRECCION = paste0(TIPOVIA, NOMVIA, NUMERO)) |> 
  select(grupo,
         tipo, 
         nombre = ETIQUETA, 
         CODMUN = CODMUNI, 
         DIRECCION, 
         X, Y,
         info) |>
  st_drop_geometry()


In [109]:
tdata_09 |> slice_head(n = 5)

grupo tipo                  nombre                           CODMUN
1 Salud Consultorios de Salud Consultorio de La Acebeda        0014  
2 Salud Consultorios de Salud Consultorio de Ajalvir           0029  
3 Salud Consultorios de Salud Consultorio de Alameda del Valle 0035  
4 Salud Consultorios de Salud Consultorio de El Álamo          0040  
5 Salud Consultorios de Salud Consultorio de Aldea del Fresno  0088  
  DIRECCION                    X         Y        info
1 PlazaSan Miguel1             -3.624803 41.08687     
2 AvdaAguas2                   -3.482595 40.53660     
3 PlazaSanta Marina17          -3.843668 40.91803     
4 CalleFederico García Lorca31 -3.988694 40.23284     
5 CalleSan Rafael17            -4.199147 40.32041

In [111]:
tdata_10 <- data_10 |>
  bind_cols(st_coordinates(data_10 |> st_transform(4326))) |> 
  mutate(tipo = "Otros Centrpos de Salud",
         grupo = "Salud",
         info = "") |> 
  select(grupo,
         tipo, 
         nombre = NOMBRE, 
         CODMUN, 
         DIRECCION, 
         X, Y,
         info) |> 
  st_drop_geometry()


In [112]:
tdata_10 |> slice_head(n = 5)

grupo tipo                   
1 Salud Otros Centrpos de Salud
2 Salud Otros Centrpos de Salud
3 Salud Otros Centrpos de Salud
4 Salud Otros Centrpos de Salud
5 Salud Otros Centrpos de Salud
  nombre                                           CODMUN
1 Instituto Provincial de Rehabilitación           079   
2 Instituto Provincial de Oftalmología             079   
3 Oficina Regional de Coordinación de Transplantes 079   
4 Oficina Regional de Coordinación de Salud Mental 079   
5 Oficina Regional de Coordinación Oncológica      079   
  DIRECCION                   X         Y        info
1 Calle Francisco Silvela, 40 -3.672685 40.43313     
2 Calle General Arrando, 17   -3.693046 40.43190     
3 Paseo de la Castellana, 280 -3.686069 40.47613     
4 Paseo de la Castellana, 280 -3.685915 40.47612     
5 Paseo de la Castellana, 280 -3.685750 40.47612

In [113]:
tdata_11 <- data_11 |>
  bind_cols(st_coordinates(data_11 |> st_transform(4326))) |> 
  mutate(tipo = "Centros de Salud Mental",
         grupo = "Salud",
         info = "",
         DIRECCION = paste0(TIPOVIAL, NOMVIA, NUMERO)) |> 
  select(grupo,
         tipo, 
         nombre = NOMBRE, 
         CODMUN, 
         DIRECCION, 
         X, Y,
         info) |>  
  st_drop_geometry()


In [114]:
tdata_11 |> slice_head(n = 5)

grupo tipo                    nombre                        CODMUN
1 Salud Centros de Salud Mental CSM Villa Vallecas            0796  
2 Salud Centros de Salud Mental CSM Coslada                   0493  
3 Salud Centros de Salud Mental CSM Salamanca                 0796  
4 Salud Centros de Salud Mental CSM Tetuán Infanto Juvenil    0796  
5 Salud Centros de Salud Mental CSM Alcalá  de Henarés I y II 0053  
  DIRECCION           X         Y        info
1 CalleSan Claudio154 -3.639029 40.38860     
2 AvenidaEspañas/n    -3.552246 40.42087     
3 CalleO'Donnell55    -3.671368 40.42154     
4 CalleAguileñas1     -3.693674 40.46551     
5 CalleOctavio Paz15  -3.372979 40.50001

In [134]:
tdata_12 <- data_12 |>
  mutate(tipo = "Hospitales",
         grupo = "Salud",
         info = paste0("CAMAS: ", CAMAS, "; ", "Clase: ", `Clase de Centro`),
         DIRECCION = Dirección,
         CODMUN = as.character(`Cód. Municipio`)) |> 
  select(grupo,
         tipo, 
         nombre = `Nombre Centro`, 
         CODMUN, 
         DIRECCION, 
         X, Y,
         info) 


In [136]:
tdata_12 |> slice_head(n = 5)

grupo tipo       nombre                                      CODMUN
1 Salud Hospitales HOSPITAL HM VALLES                          280053
2 Salud Hospitales HOSPITAL UNIVERSITARIO PRINCIPE DE ASTURIAS 280053
3 Salud Hospitales HOSPITAL QUIRONSALUD SUR                    280072
4 Salud Hospitales HOSPITAL UNIVERSITARIO FUNDACION ALCORCON   280072
5 Salud Hospitales HOSPITAL DEL TAJO S.A.                      280133
  DIRECCION                    X         Y       
1 CALLE SANTIAGO 14            -3.367372 40.48302
2 CARRETERA DE MECO S/N        -3.347602 40.51025
3 CALLE DE ESTAMBUL 30         -3.849035 40.34248
4 CALLE BUDAPEST 1             -3.837394 40.34887
5 AVENIDA AMAZONAS CENTRAL S/N -3.610795 40.05726
  info                                             
1 CAMAS: 20; Clase: Otros Centros con Internamiento
2 CAMAS: 507; Clase: Hospitales Generales          
3 CAMAS: 81; Clase: Hospitales Generales           
4 CAMAS: 402; Clase: Hospitales Generales          
5 CAMAS: 98; Clase: Hospitales Generales

In [137]:
data <-  
  bind_rows(tdata_01, tdata_02, tdata_03,
  tdata_04, tdata_05, tdata_06,
  tdata_07, tdata_08, tdata_09,
  tdata_10, tdata_11, tdata_12)

In [142]:
glimpse(data)

Rows: 1,633
Columns: 8
$ grupo     <chr> "Transporte", "Transporte", "Transporte", "Transporte", "Tra…
$ tipo      <chr> "Intercambiadores", "Intercambiadores", "Intercambiadores", …
$ nombre    <chr> "Grandes Intercambiadores Plaza Elíptica", "Grandes Intercam…
$ CODMUN    <chr> "079", "079", "079", "079", "079", "079", "079", "079", "079…
$ DIRECCION <chr> "Plaza Elíptica s/n", "Calle Princesa, 89", "Estación Prínci…
$ X         <dbl> -3.716577, -3.719508, -3.719560, -3.689044, -3.676731, -3.68…
$ Y         <dbl> 40.38540, 40.43474, 40.42080, 40.46719, 40.43797, 40.47198, …
$ info      <chr> "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", …


In [143]:
data |> slice_head(n = 5)

grupo      tipo             nombre                                     
1 Transporte Intercambiadores Grandes Intercambiadores Plaza Elíptica    
2 Transporte Intercambiadores Grandes Intercambiadores Moncloa           
3 Transporte Intercambiadores Grandes Intercambiadores Príncipe Pío      
4 Transporte Intercambiadores Grandes Intercambiadores Plaza de Castilla 
5 Transporte Intercambiadores Grandes Intercambiadores Avenida de América
  CODMUN DIRECCION                                X         Y        info
1 079    Plaza Elíptica s/n                       -3.716577 40.38540     
2 079    Calle Princesa, 89                       -3.719508 40.43474     
3 079    Estación Príncipe Pío                    -3.719560 40.42080     
4 079    Calle San Aquilino y Avenida de Asturias -3.689044 40.46719     
5 079    Avenida de América 9 A                   -3.676731 40.43797

## Synthetic Data Generation

Estos datos no requieren tareas de este tipo.

## Fake Data Generation

Estos datos no requieren tareas de este tipo.

## Open Data

Los archivos proceden de fuentes abiertas en tareas anteriores.

## <font color='green'>Data Save</font>

Este proceso, puede copiarse y repetirse en aquellas partes del notebbok que necesiten guardar datos.
Recuerde cambiar las cadenas añadida del fichero para diferenciarlas

<font color='tomato'> Identificamos los datos a guardar</font>

In [ ]:
data_to_save <- data


<font color='tomato'>Estructura de nombre de archivos:</font>

* Código del caso de uso, por ejemplo "CU_04"
* Número del proceso que lo genera, por ejemplo "_05".
* Número de la tarea que lo genera, por ejemplo "_01"
* En caso de generarse varios ficheros en la misma tarea, llevarán _01 _02 ... después
* Nombre: identificativo de "properData", por ejemplo "_zonasgeo"
* Extensión del archivo

Ejemplo: "CU_04_05_01_01_zonasgeo.json, primer fichero que se genera en la tarea 01 del proceso 05 (Data Collection) para el caso de uso 04 (vacunas)

Importante mantener los guiones bajos antes de proceso, tarea, archivo y nombre

### Proceso 05

In [138]:
caso <- "CU_18"
proceso <- '_05'
tarea <- "_02"
archivo <- ""
proper <- "_infraestructuras"
extension <- ".csv"

<font color='tomato'><b> OPCION A:</b> Uso del paquete "tcltk" para mayor comodidad</font>

* Buscar carpeta, escribir nombre de archivo SIN extensión (se especifica en el código)
* Especificar sufijo2 si es necesario
* Cambiar datos por datos_xx si es necesario

In [ ]:
# file_save <- paste0(caso, proceso, tarea, tcltk::tkgetSaveFile(), proper, extension) 
# path_out <- paste0(oPath, file_save)
# write_csv(data, path_out)

# cat('File saved as: ')
# path_out

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>

- Los ficheros de salida del proceso van siempre a Data/Output/.  

In [140]:
file_save <- paste0(caso, proceso, tarea, archivo, proper, extension) 
path_out <- paste0(oPath, file_save)
write_csv(data, path_out)

cat('File saved as: ')
path_out

File saved as: 

[1] "Data/Output/CU_18_05_02_infraestructuras.csv"

#### Copia del fichero a Input

Si el archivo se va a usar en otros notebooks, copiar a la carpeta Input

In [141]:
path_in <- paste0(iPath, file_save)
file.copy(path_out, path_in, overwrite = TRUE)

[1] TRUE

## <font color=' #2874a6 '>Main Conclusions</font>

<font color='steelblue'>List and describe the general conclusions of the analysis carried out​.</font>

### Prerequisites

This working code needs the following conditions:

* For using the interactive selection of file, the {tcltk} package must be installed. It is not needed in production.
* The {dplyr}, {readr} and {sf} packages must be installed. 
* The data paths `Data/Input` and `Data/Output` must exist (relative to the notebook path)

### Configuration Management

This notebook has been tested with the following versions of R and packages. It cannot be assured that later versions work in the same way:
* R 4.2.2
* tcltk 4.2.2
* dplyr 1.0.10
* readr 2.1.3



### Data structures

#### Objeto `data`

- Los datos de origen contienen información diversa que se ha homogeneizado en el archivo de salida
- Hay 1633 filas con las variables: 
    * grupo
	* tipo
	* nombre
	* CODMUN
	* DIRECCION
	* X
	* Y
	* info


#### Observaciones generales sobre los datos

* Las infraestructuras de metro están por "boca de metro". Al aplicar los modelos, ver si conviene agrupar por estación.
* La proyección de origen de la georeferenciación es ETRS89 (CRS 4258). En todo el proyecto se está utilizando prioritariamente WGS 84 (CRS 4326), por lo que se hace esta transformación. Si en algún momento esto cambia habría que modificar lo que corresponda en el apartado de transformación de datos

### Consideraciones para despliegue en piloto

* Los datos de origen fueron obtenidos de https://gestiona.comunidad.madrid/nomecalles/DescargaBDTCorte.icm#delimitacionesterritoriales en fichero zip con formato .shp y descomprimido en la carpeta Data/Input
* Si la información geográfica cambia, se debería actualizar el fichero y volver a ejecutar todos los procesos.

### Consideraciones para despliegue en producción

* Se deben crear los procesos ETL en producción necesarios para que los datos de entrada estén actualizados

## <font color=' #2874a6 '>Main Actions</font>

#### <font color=' #2874a6 '>Acciones done</font>

<font color='steelblue'>Indicate the actions that have been carried out in this process</font>

- Se han transformado las geometrías a WGS 84 (CRS 4326)

#### <font color=' #2874a6 '>Actions to perform</font>

<font color='steelblue'>Indicate the actions that must be carried out in subsequent processes</font>

- Asignar sección censal
- Asignar capacidad de cada infraestructura
- Obtener datos de ocupación por día de las infraestructuras
- Agregar por sección censal

## <font color='RED'>CODE TO DEPLOY (PILOT)</font>

A continuación se incluirá el código que deba ser llevado a despliegue para producción, dado que se entiende efectúa operaciones necesarias sobre los datos en la ejecución del prototipo

<font color=' red '><b>Description</b></font>

- No hay nada que desplegar en el piloto, ya que estos datos son estáticos o en todo caso cambian con muy poca frecuencia, altamente improbable durante el proyecto.


<font color=' red '><b>CODE</b></font>

In [ ]:
# incluir código